### ISSUE TITLE
Decorators on setter throws no-unsafe-any 

### ISSUE THREAD

**Issue by Melmoth-the-Wanderer** [palantir/tslint#4689]
### Bug Report

- __TSLint version__: 5.16.0
- __TypeScript version__: 3.4.3
- __Running TSLint via__: (pick one) VSCode
- Using Angular 7

#### TypeScript code being linted

```ts
    @Input() public set paramGroupId(id: string) {
        // ... some code here
    }
```

with `tslint.json` configuration:

```json
{
    "rulesDirectory": [
        "codelyzer"
    ],
    "rules": {
        "cognitive-complexity": [
            true,
            5
        ],
        "parameters-max-number": [
            true,
            5
        ],
        "no-big-function": [
            true,
            30
        ],
        "typedef": [
            true,
            "call-signature",
            "parameter"
        ],
        "only-arrow-functions": [
            true,
            "allow-named-functions"
        ],
        "adjacent-overload-signatures": true,
        "no-unsafe-any": true,
        "arrow-return-shorthand": true,
        "callable-types": true,
        "class-name": true,
        "comment-format": [
            true,
            "check-space"
        ],
        "curly": true,
        "deprecation": {
            "severity": "warn"
        },
        "eofline": true,
        "forin": true,
        "import-blacklist": [
            true,
            "rxjs/Rx"
        ],
        "import-spacing": true,
        "indent": [
            true,
            "spaces",
            4
        ],
        "interface-over-type-literal": true,
        "label-position": true,
        "max-line-length": [
            true,
            140
        ],
        "member-access": true,
        "member-ordering": [
            true,
            {
                "order": [
                    "public-static-field",
                    "public-instance-field",
                    "protected-static-field",
                    "protected-instance-field",
                    "private-static-field",
                    "private-instance-field",
                    "constructor",
                    "static-method",
                    "instance-method"
                ]
            }
        ],
        "no-arg": true,
        "no-bitwise": true,
        "no-console": [
            false,
            "debug",
            "info",
            "time",
            "timeEnd",
            "trace"
        ],
        "no-construct": true,
        "no-debugger": true,
        "no-duplicate-super": true,
        "no-duplicate-string": [
            true,
            3
        ],
        "no-empty": false,
        "no-empty-interface": true,
        "no-eval": true,
        "no-inferrable-types": [
            true,
            "ignore-params"
        ],
        "no-misused-new": true,
        "no-non-null-assertion": true,
        "no-redundant-jsdoc": true,
        "no-shadowed-variable": true,
        "no-string-literal": false,
        "no-string-throw": true,
        "no-switch-case-fall-through": true,
        "no-trailing-whitespace": true,
        "no-unnecessary-initializer": true,
        "no-unused-expression": true,
        "no-use-before-declare": true,
        "no-var-keyword": true,
        "object-literal-sort-keys": [
            true,
            "ignore-case",
            "match-declaration-order"
        ],
        "one-line": [
            true,
            "check-open-brace",
            "check-catch",
            "check-whitespace"
        ],
        "prefer-const": true,
        "quotemark": [
            true,
            "single"
        ],
        "radix": true,
        "semicolon": [
            true,
            "always"
        ],
        "triple-equals": [
            true
        ],
        "typedef-whitespace": [
            true,
            {
                "call-signature": "nospace",
                "index-signature": "nospace",
                "parameter": "nospace",
                "property-declaration": "nospace",
                "variable-declaration": "nospace"
            },
            {
                "call-signature": "onespace",
                "index-signature": "onespace",
                "parameter": "onespace",
                "property-declaration": "onespace",
                "variable-declaration": "onespace"
            }
        ],
        "unified-signatures": true,
        "variable-name": [
            true,
            "check-format",
            "allow-leading-underscore",
            "require-const-for-all-caps",
            "allow-pascal-case",
            "ban-keywords"
        ],
        "whitespace": [
            true,
            "check-branch",
            "check-decl",
            "check-operator",
            "check-module",
            "check-separator",
            "check-rest-spread",
            "check-type",
            "check-typecast",
            "check-type-operator",
            "check-preblock"
        ],
        "no-output-on-prefix": true,
        "use-input-property-decorator": true,
        "use-output-property-decorator": true,
        "use-host-property-decorator": true,
        "no-input-rename": true,
        "no-output-rename": true,
        "use-life-cycle-interface": true,
        "use-pipe-transform-interface": true,
        "component-class-suffix": true,
        "directive-class-suffix": true
    }
}
```

#### Actual behavior
tslint reports a break of rule "Unsafe use of expression of type 'any'. (no-unsafe-any)tslint(1)". The `Input()` statement is marked as faulty. It works fine for fields but reports an error when used on a setter.

#### Expected behavior
No problems reported on the decorator used on a setter/method.

**Comment by Melmoth-the-Wanderer** [comment]
Looks like a dup #3784. I failed searching it at the first time.

**Comment by Melmoth-the-Wanderer** [comment]
The requested change in angular project is reported here https://github.com/angular/angular/issues/27595
